Mathlib/RingTheory/Bialgebra/Hom.lean

```diff
@@ -155,19 +155,13 @@ theorem coe_coalgHom_mk {f : A →ₗc[R] B} (h h₁) :
     ((⟨f, h, h₁⟩ : A →ₐc[R] B) : A →ₗc[R] B) = f := by
   rfl
 
-@[simp, norm_cast]
-theorem coe_toCoalgHom (f : A →ₐc[R] B) : ⇑(f : A →ₗc[R] B) = f :=
-  rfl
+@[simp, norm_cast] lemma coe_toCoalgHom (f : A →ₐc[R] B) : ⇑(f : A →ₗc[R] B) = f := rfl
+@[norm_cast] lemma coe_toAlgHom (f : A →ₐc[R] B) : ⇑(f : A →ₐ[R] B) = f := by simp
 
-lemma toCoalgHom_apply (f : A →ₐc[R] B) (a : A) : f.toCoalgHom a = f a := rfl
+@[deprecated "Use the `CoalgHom` API." (since := "2025-08-10")]
```

Comment:
I don't understand. Could you explain more? In particular, which lemma are you referring to?

Reply:
I think the only reason why it is currently not `BialgHom`'s job is because we have the bad coercion `SemilinearMapClass.semilinearMap` but morally I would argue this should be here (see e.g. `AlgHom.coe_toRingHom` and `DistribMulActionHom.coe_toLinearMap` and `CoalgHom.coe_toLinearMap` and more). I think this is also making the future work of removing `SemilinearMapClass.semilinearMap` more difficult.

Reply:
Indeed, I have already thought about this and my reasoning goes further:

Once we remove `SemilinearMapClass.semilinearMap`, we will want to have two coercions from `BialgHom`: one to `AlgHom` and one to `CoalgHom`. We do not want to add a direct coercion to `LinearMap`, because we would be back to having quadratically many coercions. Instead, we want to make one of the two paths `BialgHom -> AlgHom -> LinearMap` and `BialgHom -> CoalgHom -> LinearMap` have higher priority (probably the one through `AlgHom`) and use the transitive coercion (which is syntactically the composition of its two constituent coercions).

Therefore we will not need `BialgHom.coe_toLinearMap`, and I am not making the future refactor more difficult.

Reply:
No? It's a simp lemma, people should not have to call it by hand (and indeed it was never called by hand). Rewriting backwards with such a lemma is rare enough that we shouldn't worry about it.